src/components/Courseware/CoursewareNav/index.jsx

```diff
@@ -46,8 +46,18 @@ class CoursewareNav extends React.Component {
   }
 }
 
+const nodeShape = PropTypes.shape({
+  displayName: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  descendants: PropTypes.array.isRequired,
+});
+
 CoursewareNav.propTypes = {
-  courseOutline: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+  courseOutline: PropTypes.shape({
+    displayName: PropTypes.string.isRequired,
+    descendants: PropTypes.arrayOf(nodeShape).isRequired,
```

Comment:
Is there ever a case where the `courseOutline` prop will have more than one level of `descendants`? If that is possible, can this `courseOutline` prop handle multiple levels?

Reply:
Yes, that is supported because `nodeShape` also contains descendants. 

Ideally `nodeShape.descendants` would be an `arrayOf(nodeShape)`  Not able to do that in a single declaration above but could do it in 2 lines. Opted to leave it as a generic array because I got scared off by the comment recursive propTypes could be "[arbitrarily slow](https://github.com/facebook/react/issues/5676)"  and didn't see a lot of value but if we think that level of checking is necessary I can update.